rfcs/97-add-maximum-file-size-for-storage-write.md

```diff
@@ -0,0 +1,71 @@
+---
+author: JinnyYi <github.com/JinnyYi>
+status: draft
+updated_at: 2021-06-09
+---
+
+# GSP-97: Add Maximum File Size for Storage Write
+
+## Background
+
+Depending on the size of the data you are uploading, services offer different options, like upload a file in a single operation, in parts, or through append operation.
+
+We use `Write` to handle single write operation. And the size is not unlimited.
+
+- S3 alike storage services have 5GB limit for a single PUT operation.
+- azblob limited to 5000 MiB <https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob>
+
+If we upload a file with size out of limit in a single operation, we will get an error like `Request Entity Too Large` in azblob. Same with `copy`,`fetch`, etc.
+
+We should figure out the error before sending request to reduce the consumption of network resources.
+
+## Proposal
+
+So I propose to add maximum file size for storage write related operations.
+
+A storager will be accompanied by the settings after initialization. We could add a variable with type `interface{}` into the current `Storage` structure in services to hold these settings and possible limitations in the future.
```

Comment:
How about adding this struct in `StorageMeta` instead of `Storage`?

Reply:
Can we add this struct in `StorageMeta` then add a member with the type `StorageMeta` and remove the `name`, `workDir` in `Storage`?

Reply:
LGTM